portable/ThirdParty/GCC/Posix/port.c

```diff
@@ -430,6 +448,9 @@ Thread_t *pxThread = pvParams;
 	/* Call the task's entry point. */
 	pxThread->pxCode( pxThread->pvParams );
 
+	/* Task function should not return */
+	prvTaskExitError();
```

Comment:
I think the normal way to handle asserts is like this https://github.com/FreeRTOS/FreeRTOS/blob/6cc5310f380524c7885181a3be0c60ead4b59a22/FreeRTOS/Demo/Posix_GCC/main.c#L283

That means you probably just have to call configASSERT( pdFALSE );  over here and be done.

Reply:
@cobusve This is exactly what `prvTaskExitError();` is doing in addition to stop if `configASSERT()` is not actually configured. This is useful to cover the issue of task ending in FreeRTOS and be easy to catch.
The same function is used in many ports (For example https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/master/portable/GCC/ARM_CM3/port.c#L194)

Reply:
In some port it is defined by the demo/application, and in some ports it is defined by the port
in this case this function is defined by the demo application in: https://github.com/FreeRTOS/FreeRTOS/blob/master/FreeRTOS/Demo/Posix_GCC/main.c#L283

Reply:
Sorry I did not get what is the issue here. If Posix port **can** be used without defining `configASSERT` then the stop loop is needed. If Posix port **can not** be used without defining `configASSERT` then no need for the stop loop but we need to write this down in some documentation.

Reply:
https://www.freertos.org/a00110.html#configASSERT
As you can see in the doc, config assert is an application side setting not a port side setting, as the user has the option of turning it on or off depending on the maturity of the application in order to save space.

Reply:
This is exactly what I am talking about. So that the stop loop in `prvTaskExitError` after config assert is important to avoid dead silent stop when config assert is not set in the application side

Reply:
If the user is confident about his application, and is sure that they called vTaskDelete at the end of every task, extra code and code size is not desired, so vConfigAssert will rightfully be disabled to remove the extra code. By adding the prvTaskExitError function this would not be possible

Worth mentioning, this is a linux port and extra code is not as important, but to stay consistent, with other new ports,  be as true as possible to an embedded environment, and provide a good example.